codice/ddf
# [2.26.x] DDF-6636 adds IP and port to all audit events

#### What does this PR do?
This PR adds the client IP and Port to all audit events.  It does this by utilizing the client-info map in the ThreadContext.  Prior to this, the IP/port would only get logged if the audit event included a CXF message.  This wasn't present in many cases (Websocket requests and other non-CXF requests). 


#### Who is reviewing it? 
@glenhein @SmithJosh @derekwilhelm @pklinef 

#### Select relevant component teams: 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.
Any of @glenhein @SmithJosh @derekwilhelm @pklinef 

#### How should this be tested?
- Build/install and add ddf-ui features
- Generate audit events in a variety of ways
- [ ] Search from the simple UI 
- [ ] Search from the catalog ui (using websockets)
- [ ] Upload from the UI
- [ ] Download from the UI
- [ ] edit a metacard from the UI
- [ ] delete a metacard from the UI
- [ ] Update a configuration from the Admin UI
- [ ] submit requests directly to an endpoint (like the REST endpoint)

Verify that the audit events include the Client IP/Port in all cases.

#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: #6636 

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
